src/pager.c

```diff
@@ -697,8 +697,9 @@ struct Pager {
   char *pTmpSpace;            /* Pager.pageSize bytes of space for tmp use */
   PCache *pPCache;            /* Pointer to page cache object */
 #ifndef SQLITE_OMIT_WAL
-  Wal *pWal;                  /* Write-ahead log used by "journal_mode=wal" */
-  char *zWal;                 /* File name for write-ahead log */
+  Wal *pWal;                       /* Write-ahead log used by "journal_mode=wal" */
+  libsql_wal_methods *pWalMethods; /* Virtual methods for interacting with WAL */
```

Comment:
Why do we need to store pWalMethods here when pWal already has them in pMethods field ?

Reply:
Related to the comment above - Wal used to (and I'm pretty sure still should be) opaque to upper layers, pager included. And once `struct Wal` is opaque, we need to somehow propagate WAL methods from `libsql_open` down to WAL, and pager is the only way through.

Reply:
as above - struct Wal is still opaque to its upper layers (pager), but it's not opaque for the authors of custom virtual WAL implementations